sdks/python/apache_beam/transforms/display/display_data.py

```diff
@@ -0,0 +1,133 @@
+"""
+DisplayData, its classes, interfaces and methods.
+"""
+
+from __future__ import absolute_import
+
+import calendar
+from datetime import datetime, timedelta
+import inspect
+import json
+
+__all__ = ['HasDisplayData', 'DisplayDataItem', 'DisplayData']
+
+class HasDisplayData(object):
+  """ Basic interface for elements that contain display data.
+
+  It contains only the display_data method.
+  """
+  def __init__(self, *args, **kwargs):
+    super(HasDisplayData, self).__init__(*args, **kwargs)
+
+  def display_data(self):
+    return {}
+
+  def _namespace(self):
+    return '{}.{}'.format(self.__module__, self.__class__.__name__)
+
+
+class DisplayData(object):
+  def __init__(self, namespace='__main__'):
+    self.namespace = namespace
+    self.items = []
+
+  def populate_items(self, display_data_dict):
+    for key, element in display_data_dict.items():
+      if isinstance(element, HasDisplayData):
+        subcomponent_display_data = DisplayData(element._namespace())
+        subcomponent_display_data.populate_items(element.display_data())
+        self.items += subcomponent_display_data.items
+        continue
+
+      if isinstance(element, dict):
```

Comment:
I feel strongly we shouldn't prohibit ourselves from ever allowing dict as a value in the future. Also, it's more pythonic to take a dict when the key set is more free-form, or user-defined, but where you have this structure here it'd be better to take a (documented) DisplayDataItem object itself rather than a dict whose named members are used to create this dict (and whose other elements are silently ignored).